java/opendcs/src/main/resources/db/CWMS-Oracle/callbacks/beforeMigrate__set_schema_permissions.sql

```diff
@@ -0,0 +1,65 @@
+grant cwms_user to ${CCP_SCHEMA};
+-- example below from https://stackoverflow.com/a/1199438
+begin
+  execute immediate 'create role ${CCP_SCHEMA}_users';
+exception
+  when others then
+    --"ORA-01921: role name 'x' conflicts with another user or role name"
+    if sqlcode = -01921 then 
+      null;
+    else
+      raise;
+    end if;
+end;
+/
+
+grant create session,resource,connect to ${CCP_SCHEMA}_users;
+grant ${CCP_SCHEMA}_USERS to cwms_user;
+GRANT  ALTER ANY TABLE,CREATE ANY TABLE,CREATE ANY INDEX,CREATE ANY SEQUENCE,
+      CREATE ANY VIEW,CREATE ANY PROCEDURE,CREATE ANY TRIGGER,CREATE ANY JOB,
+      CREATE ANY SYNONYM,DROP ANY SYNONYM,CREATE PUBLIC SYNONYM,DROP PUBLIC SYNONYM
+    TO ${CCP_SCHEMA};
+
+GRANT CREATE ANY CONTEXT TO ${CCP_SCHEMA};
+GRANT SELECT ON dba_scheduler_jobs to ${CCP_SCHEMA};
+GRANT SELECT ON dba_queue_subscribers to ${CCP_SCHEMA};
+GRANT SELECT ON dba_subscr_registrations to ${CCP_SCHEMA};
+GRANT SELECT ON dba_queues to ${CCP_SCHEMA};
+GRANT EXECUTE ON dbms_aq TO ${CCP_SCHEMA};
+GRANT EXECUTE ON dbms_aqadm TO ${CCP_SCHEMA};
+GRANT EXECUTE ON DBMS_SESSION to ${CCP_SCHEMA};
+GRANT EXECUTE ON DBMS_RLS to ${CCP_SCHEMA};
```

Comment:
Other parts of the code have  suffix '_users'    , but why not on these grants?

I see one example below where both with and without the '_users' is used.

${CWMS_SCHEMA}.cwms_env

Reply:
These grants are specific to the CCP schema "user". Their uses is through the CCP procedures which handle additional permissions checks.

the ${CCP_SCHEMA}_users is the name of the role granted to anyone who has an account and will use CCP.